polygeist/tools/cgeist/Test/Verification/sycl/math_funcs.cpp

```diff
@@ -0,0 +1,86 @@
+// RUN: clang++ -Xcgeist --use-opaque-pointers=1 -fsycl -fsycl-targets=spir64-unknown-unknown-syclmlir -O0 -w %s -o %t.O0.out 2>&1 | FileCheck %s --allow-empty --implicit-check-not="{{error|Error}}:"
+// RUN: clang++ -Xcgeist --use-opaque-pointers=1 -fsycl -fsycl-targets=spir64-unknown-unknown-syclmlir -O1 -w %s -o %t.O1.out 2>&1 | FileCheck %s --allow-empty --implicit-check-not="{{error|Error}}:"
+// RUN: clang++ -Xcgeist --use-opaque-pointers=1 -fsycl -fsycl-targets=spir64-unknown-unknown-syclmlir -O2 -w %s -o %t.O2.out 2>&1 | FileCheck %s --allow-empty --implicit-check-not="{{error|Error}}:"
+// RUN: clang++ -Xcgeist --use-opaque-pointers=1 -fsycl -fsycl-targets=spir64-unknown-unknown-syclmlir -O3 -w %s -o %t.O3.out 2>&1 | FileCheck %s --allow-empty --implicit-check-not="{{error|Error}}:"
+// RUN: clang++ -Xcgeist --use-opaque-pointers=1 -fsycl -fsycl-targets=spir64-unknown-unknown-syclmlir -Ofast -w %s -o %t.Ofast.out 2>&1 | FileCheck %s --allow-empty --implicit-check-not="{{error|Error}}:"
+
+// RUN: clang++ -Xcgeist --use-opaque-pointers=1 -fsycl -fsycl-device-only -O0 -w -emit-llvm -fsycl-targets=spir64-unknown-unknown-syclmlir %s -o %t.bc
+
+// Test that the LLVMIR generated is verifiable.
+// RUN: opt -passes=verify -disable-output < %t.bc
+
+// Verify that LLVMIR generated is translatable to SPIRV.
+// RUN: llvm-spirv %t.bc
+
+// Test that the SYCL math functions are lowered to llvm intrinsics.
+// RUN: llvm-dis %t.bc
+// RUN: cat %t.ll | FileCheck %s --check-prefix=LLVM
+
+#include <climits>
+#include <cmath>
+
+#include <sycl/sycl.hpp>
+using namespace sycl;
+
+void host_math_funcs(std::array<float, 1> &A) {
+  auto q = queue{};
+  device d = q.get_device();
+  std::cout << "Using " << d.get_info<info::device::name>() << "\n";
+
+  {
+    auto buf = buffer<float, 1>{A.data(), 1};
+    q.submit([&](handler &cgh) {
+      auto A = buf.get_access<access::mode::read_write>(cgh);
+      cgh.single_task<class kernel_math_funcs>([=]() {
```

Comment:
We might not need a full program here. Maybe we can have:

```cpp
SYCL_EXTERNAL float f(float a, float b, float c) {
   ...
}
```

Reply:
Yeah makes sense, I'll look into that.